ff4j-web/src/main/java/org/ff4j/web/taglib/AbstractFeatureTag.java

```diff
@@ -109,8 +111,52 @@ else if (eval(ff4j, pageContext)) {
 
         return result ? EVAL_BODY_INCLUDE : SKIP_BODY;
     }
-    
-    protected abstract boolean eval(FF4j ff4j, PageContext pageContext);
+
+    protected boolean eval(FF4j ff4j, PageContext pageContext) {
+        FlippingExecutionContext currentContext = ff4j.getCurrentContext();
+        FlippingExecutionContext executionContext = isShareHttpSession() ? createExecutionContextFromSharedHttpSession(pageContext, currentContext) : currentContext;
+
+        try {
+            return evalWithExecutionContext(ff4j, pageContext, executionContext);
+        } finally {
+            // Restore original current context, otherwise some parameters from previous JSP context remains.
+            if (isShareHttpSession()) {
+                ff4j.setCurrentContext(currentContext);
```

Comment:
Here, we restore the original current context, otherwise we may keep some undesirable variables for the next request (that were added due to the request parameter map).

Reply:
In fact, I'm not sure to understand why the execution context is set in the check method, but I have no doubt there is a good reason for that 😉